x-pack/plugins/monitoring/server/lib/alerts/fetch_cpu_usage_node_stats.ts

```diff
@@ -179,14 +179,17 @@ async function fetchContainerStats(
       }
 
       const limitsNotSet = node.quota_micros_max.value === -1 && node.quota_micros_min.value === -1;
-      const notRunningInAContainer =
-        node.quota_micros_min.value === null && node.quota_micros_max.value === null;
-      if (limitsNotSet || notRunningInAContainer) {
+      if (limitsNotSet) {
+        const cpuUsage = node.average_cpu_usage_percent.value ?? undefined;
+
+        logger.warn(
+          `CPU usage rule: Node "${node.key}" does not have resource limits configured. Fallback metric reports usage of ${cpuUsage}%.`
+        );
+
```

Comment:
I'm having real doubts about if we should log something or not. 

On the one hand, this looks to _**me**_ as a missed configuration but maybe it's more common than I think? 
And then, having this logged every time the rule executes, possibly for many nodes, feels like a risk for spam which will make the Kibana log grow which means more cost of storing that log for no real gain and the customer might be fine with it. 

So maybe we should make it a debug level instead? But, if we do, is it really useful? Will it help customers notice something is wrong.
I don't know if users really look at the Kibana logs anyway, similar for the log we make below about values missing?

Maybe both of those cases should just rely on the fallback metric silently? 

I just feel like in the last PR, I included more changes than needed for the bug fix.
I think the limits change part is good because that's a real cause for bugs in the calculation. 

But the other two cases should probably just silently return the base CPU metric since I'm not sure if either really helps the user find issues?

So my vote is to remove the logging here, and combine this check with the check below about if they are null, and not report any special status about that. What do you think?
That would bring the total set of changes closer to fixing only the original problem and removes a bunch of noise from the logs.

Reply:
```
const limitsNotSet = node.quota_micros_max.value === -1 && node.quota_micros_min.value === -1;
// This check was mostly for the compiler to be happy about the types anyway
const statsMissing =  node.max_usage_nanos.value === null &&
        node.min_usage_nanos.value === null &&
        node.max_periods.value === null &&
        node.min_periods.value === null &&
        node.quota_micros_max.value === null;

if (limitsNotSet || statsMissing) {
  const cpuUsage = node.average_cpu_usage_percent.value ?? undefined;
  return {
  clusterUuid: cluster.key as string,
  nodeId: node.key as string,
  cpuUsage,
  nodeName,
  ccs,
  };
}
```

Then after we check if the values are different or not.